libr/asm/asm.c

```diff
@@ -198,6 +198,9 @@ R_API RAsm *r_asm_free(RAsm *a) {
 			r_list_free (a->plugins);
 			a->plugins = NULL;
 		}
+		if (a->syscall) {
```

Comment:
unnecessary if

Reply:
`a->syscall` may be `NULL`

The duplicate of `RAsm::syscall` and `RAnal::syscall` looks suspicious.

Reply:
there's no problem on calling r_syscall_free(NULL); so remove the if